### PR TITLE
feat: filters out centos systems

### PR DESCRIPTION
**Pull request by petrblaho**
RHINENG-2630

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: kerlescan/inventory_service_interface.py [hunk 1]
```diff
@@ -141,6 +141,15 @@ def fetch_systems_with_profiles(system_ids, service_auth_key, logger, counters):
...
-    return interleave_systems_and_profiles(
+    systems_with_profiles = interleave_systems_and_profiles(
         systems_result, system_profiles_result, system_tags_result
     )
+
+    # remove CentOS Linux systems
+    systems_with_profiles = [
+        system
+        for system in systems_with_profiles
+        if system["operating_system"]["name"] == "CentOS Linux"
```
Comment (vkrizan) [inline kerlescan/inventory_service_interface.py:152]: constant

File: kerlescan/inventory_service_interface.py [hunk 2]
```diff
@@ -141,6 +141,15 @@ def fetch_systems_with_profiles(system_ids, service_auth_key, logger, counters):
 
     ensure_correct_system_count(system_ids, systems_result)
 
-    return interleave_systems_and_profiles(
+    systems_with_profiles = interleave_systems_and_profiles(
         systems_result, system_profiles_result, system_tags_result
     )
+
+    # remove CentOS Linux systems
```
Comment (vkrizan) [inline kerlescan/inventory_service_interface.py:148]: The below code selects and not deletes, as this comment explains.

Comment (petrblaho) [reply kerlescan/inventory_service_interface.py:149]: Well, it's Friday... thank you

File: kerlescan/inventory_service_interface.py [hunk 3]
```diff
@@ -141,6 +142,15 @@ def fetch_systems_with_profiles(system_ids, service_auth_key, logger, counters):
...
+    systems_with_profiles = [
+        system
+        for system in systems_with_profiles
+        if system["operating_system"]["name"] not in FILTERED_OUT_OPERATING_SYSTEMS
```
Comment (vkrizan) [inline kerlescan/inventory_service_interface.py:153]: Is `name` always present in the `system["operating_system"]`?

Comment (petrblaho) [reply kerlescan/inventory_service_interface.py]: no, you are right, also `operating_system` is not always present.....

solved...

File: kerlescan/inventory_service_interface.py [hunk 4]
```diff
@@ -27,6 +28,17 @@ def ensure_correct_system_count(system_ids_requested, result):
         raise ItemNotReturned("ids [%s] not available to display" % ", ".join(missing_ids))
 
 
+def filter_out_systems(systems_with_profiles):
+    # remove CentOS Linux systems
+    systems_with_profiles = [
```
Comment (vkrizan) [inline kerlescan/inventory_service_interface.py:33]: Nit: can do a return here.

**Review by vkrizan — APPROVED** [review]

File: kerlescan/inventory_service_interface.py (hunk 4 above)
Comment (petrblaho) [reply kerlescan/inventory_service_interface.py:33]: I know :-)

**Pull request merged**